bridgecrew_sns_resource.tf

```diff
@@ -0,0 +1,57 @@
+data template_file "message" {
+  template = file("${path.module}/sns_message.json")
+  vars = {
+    request_type         = "Create"
+    bridgecrew_sns_topic = local.sns_topic_arn
+    account_id = local.account_id
+    template_version = local.template_version
+    inbound_sqs = aws_sqs_queue.InboundRemediations.id
+  }
+}
+
+resource "null_resource" "await_resources" {
+  provisioner "local-exec" {
+    command = "sleep 10"
+  }
+
+  depends_on = [aws_sqs_queue.InboundRemediations, aws_sqs_queue.RemediationsDLQ, aws_lambda_function.BridgecrewRemoteRemediation, aws_iam_role.RemediationFunctionRole]
+}
+
+resource null_resource "create_bridgecrew" {
+  provisioner "local-exec" {
+    command     = "aws sns ${local.profile_str} --region ${local.region} publish --target-arn \"${local.sns_topic_arn}\" --message '${jsonencode(data.template_file.message.rendered)}' && sleep 30"
+    working_dir = path.module
+  }
+
+  depends_on = [null_resource.await_resources]
+}
+
+resource null_resource "update_bridgecrew" {
+  triggers = {
+    build = md5(data.template_file.message.rendered)
+  }
+
+  provisioner "local-exec" {
+    command     = "aws sns ${local.profile_str} --region ${local.region} publish --target-arn \"${local.sns_topic_arn}\" --message '${jsonencode(replace(data.template_file.message.rendered, "Create", "Update"))}'"
+    working_dir = path.module
+  }
+
+  depends_on = [null_resource.create_bridgecrew]
+}
+
+resource null_resource "disconnect_bridgecrew" {
+  triggers = {
+    profile   = local.profile_str
+    region    = local.region
+    message   = jsonencode(replace(data.template_file.message.rendered, "Create", "Delete"))
+    sns_topic = local.sns_topic_arn
+  }
+
+  provisioner "local-exec" {
+    command     = "aws sns ${self.triggers.profile} --region ${self.triggers.region} publish --target-arn \"${self.triggers.sns_topic}\" --message '${self.triggers.message}'"
+    when        = destroy
+    working_dir = path.module
+  }
+
+  depends_on = [null_resource.create_bridgecrew]
+}
```

Comment:
We have to get an SNS message in order to process the integration. How would you propose doing it?

Not sure I got the "install and run the pre-commits" bit.

Reply:
In the root of the repo is the pre-commit config. Install pre-commit from pip, then pre-commit install in your shell at the root of your repo. Ever commit then has to pass you pre-commit before it can be pushed to git. In this case it checks for secrets, formatting of tf,json and your readme and also runs checkov.

Reply:
id not use tf for invoking it at all, its a shell script.

Reply:
But how do you want to get the user to run the script / command to notify us of the integration when they add or delete it?

Reply:
pre commit is more for the developer that is writing the module. it helps with terraform fmt and running checkov locally. @JamesWoolfenden can you run it and add additional commits to demo the change?